python/lsst/skymap/packers.py

```diff
@@ -60,56 +144,163 @@ def getIntFromFilter(cls, name):
             raise NotImplementedError(f"band '{name}' not supported by this ID packer.")
 
     @classmethod
+    @deprecated(
+        reason="This classmethod cannot reflect all __init__ args and will be removed after v27.",
+        version="v26.0",
+        category=FutureWarning,
+        action="error",
+    )
     def getFilterNameFromInt(cls, num):
-        """Return an band name from its integer representation.
-        """
+        """Return an band name from its integer representation."""
         return cls.SUPPORTED_FILTERS[num]
 
     @classmethod
+    @deprecated(
+        reason="This classmethod cannot reflect all __init__ args and will be removed after v27.",
+        version="v26.0",
+        category=FutureWarning,
+        action="error",
+    )
     def getMaxIntForFilters(cls):
         return len(cls.SUPPORTED_FILTERS)
 
-    @classmethod
-    def configure(cls, dimensions):
-        # Docstring inherited from DataIdPacker.configure
-        assert dimensions.given == ["skymap"]
-        assert dimensions.required.issuperset(["tract", "patch"])
-        metadata = {"skymap": ["tract_max", "patch_nx_max", "patch_ny_max"]}
-        kwds = {}
-        return metadata, kwds
-
-    def __init__(self, fixed: DataCoordinate, dimensions: DimensionGraph):
-        super().__init__(fixed, dimensions)
-        record = fixed.records["skymap"]
-        self._skyMapName = record.name
-        self._patchMax = record.patch_nx_max * record.patch_ny_max
-        self._tractPatchMax = self._patchMax*record.tract_max
-        if "band" in dimensions:
-            self._filterMax = self.getMaxIntForFilters()
+    def __init__(
+        self,
+        fixed: DataCoordinate,
+        dimensions: DimensionGraph | None = None,
+        bands: Mapping[str, int] | None = None,
+        n_bands: int | None = None,
+        n_tracts: int | None = None,
+        n_patches: int | None = None,
+    ):
+        if bands is None:
+            bands = {b: i for i, b in enumerate(self.SUPPORTED_FILTERS)}
+        if dimensions is None:
+            if n_bands is None:
+                n_bands = len(bands)
+            dimension_names = ["tract", "patch"]
+            if n_bands != 0:
+                dimension_names.append("band")
+            dimensions = fixed.universe.extract(dimension_names)
         else:
-            self._filterMax = None
+            if "band" not in dimensions.names:
+                n_bands = 0
```

Comment:
Should the code also raise here if `bands` is not `{}` or `None` and/or `n_bands` is not `0` or `None`? Ensuring people aren't passing in inconsistent kwargs might be good.

Reply:
Outside unit tests I expect this class to be constructed pretty much only via config, so satisfying those checks would mean a lot more analogous logic in `from_config` just to satisfy them, and hence I think the overall gain in maintainability would be small or negative.